### PR TITLE
feat(lint): add linting to universal starter

### PR DESCRIPTION
**Pull request by CaerusKaru**
* Add linting with parity to defaults provided by Angular CLI
* Refactor existing code to conform with linting

### CODE REVIEW

**Comment by CaerusKaru** [comment]
cc @vikerman @Toxicable

**Comment by Toxicable** [comment]
I think a pre-commit hook https://www.npmjs.com/package/pre-commit + `ng lint --fix` would improve this a bit.
Just trying to get access back on this repo and we'll get it merged

**Comment by CaerusKaru** [comment]
@Toxicable I changed the lint command from `ng lint` to `ng lint --fix` and added the lint step to the precommit array

File: .gitignore [hunk 1]
```diff
@@ -2,6 +2,7 @@
 .DS_Store
 .vscode
 morgan.log
+package-lock.json
```
Comment (Toxicable) [inline .gitignore]: same as the other issue, this shoudn't be ignored

File: tslint.json [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+      "check-branch",
+      "check-decl",
+      "check-operator",
+      "check-separator",
+      "check-type"
+    ],
+    "directive-selector": [
+      true,
+      "attribute",
+      "<%= prefix %>",
```
Comment (Toxicable) [inline tslint.json]: should be a prefix here, lets use `app` since that's the default for a cli projecty

Comment (CaerusKaru) [reply tslint.json]: Great catch, fixed

File: .gitignore (hunk 1 above)
Comment (CaerusKaru) [reply .gitignore]: Fixed

**Pull request merged**